### PR TITLE
Add a Daml Hub deploy script

### PR DESCRIPTION
**Pull request by gaborh-da**

### CODE REVIEW

File: scripts/dabl_deploy.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+#
+
+
+# Terminate on any failure
+set -ex
+
+STAMP=$(date "+%H%M%S")
+PROJECT_NAME=Cbdc$STAMP
+
+LEDGER_NAME=cbdc
```
Comment (richardkapolnai-da) [inline scripts/dabl_deploy.sh:14]: just wondering: shouldn't the ledger name be unique as project name?

Comment (gaborh-da) [reply scripts/dabl_deploy.sh:14]: I think it does not make a difference. I mean, I deployed 10-20 projects with ledgers (having the same name) after each other and they did not bother each other.

File: scripts/dabl_deploy.sh [hunk 2]
```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/env bash
+#
+# Copyright (c) 2019, Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+
+
+# Terminate on any failure
```
Comment (richardkapolnai-da) [inline scripts/dabl_deploy.sh:8]:
```suggestion
```

File: scripts/dabl_deploy.sh [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+set -ex
+
+STAMP=$(date "+%H%M%S")
+PROJECT_NAME=Cbdc$STAMP
+
+LEDGER_NAME=cbdc
+# The 'jq -r' removes the superfluous quotes for our variables.
```
Comment (richardkapolnai-da) [inline scripts/dabl_deploy.sh:15]:
```suggestion
```

File: scripts/dabl_deploy.sh [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+PROJECT_ID=`dablc -j project ensure ${PROJECT_NAME} | jq -r '.project_id'`
+LEDGER_ID=`dablc -j ledger create ${PROJECT_ID} ${LEDGER_NAME} | jq -r '.ledger_id'`
+
+echo $LEDGER_ID
```
Comment (richardkapolnai-da) [inline scripts/dabl_deploy.sh:20]:
```suggestion
echo "Deploying $PROJECT_ID / $LEDGER_ID"
```

File: scripts/dabl_deploy.sh [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+# Create Users on ledger.
```
[the remark this answers is not in the document]
Comment (richardkapolnai-da) [reply scripts/dabl_deploy.sh:22]: please get rid of all comments, and only keep which are absolutely necessary.

File: scripts/dabl_deploy.sh [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+TRIGGER_HASH=`dablc -j workspace install triggers-1.0.0.dar ${LEDGER_ID} | jq -r '.artifact_hash'`
+
+# Dar deploy.
+dablc -j ledger dar ${LEDGER_ID} ${DAR_SHA}
+
+# UI deploy.
+dablc -j ledger ui ${LEDGER_ID} ${UI_SHA}
+
+# mac os specific
+sleep 120
```
Comment (richardkapolnai-da) [inline scripts/dabl_deploy.sh:58]: untested:
```suggestion
function sleep_because_macos_needs_it_for_unknown_reasons() {
  sleep 120
}

sleep_because_macos_needs_it_for_unknown_reasons
```

Comment (gaborh-da) [reply scripts/dabl_deploy.sh:58]: okay

File: scripts/dabl_deploy.sh (hunk 5 above)
[the remark this answers is not in the document]
Comment (gaborh-da) [reply scripts/dabl_deploy.sh:22]: okay

File: scripts/dabl_deploy.sh (hunk 4 above)
Comment (gaborh-da) [reply scripts/dabl_deploy.sh:20]: okay

File: scripts/dabl_deploy.sh (hunk 3 above)
Comment (gaborh-da) [reply scripts/dabl_deploy.sh:15]: I would rather leave this, this is the only comment helpful for me.

File: scripts/dabl_deploy.sh (hunk 2 above)
Comment (gaborh-da) [reply scripts/dabl_deploy.sh:8]: yes

File: scripts/dabl_deploy.sh (hunk 3 above)
Comment (richardkapolnai-da) [reply scripts/dabl_deploy.sh]: doesn't `jq -h` or `man jq` gives you the same info?
we shouldn't include jq doc here imho

Comment (richardkapolnai-da) [reply scripts/dabl_deploy.sh]: if you really need it, perhaps:
```
alias jq_rawstring='jq -r'
```
and use it everywhere.
still sounds a bit redundant, but acceptable.

**Comment by richardkapolnai-da** [comment]
and please attend the warnings about Bash, e.g. replace all `${VAR}` with `"$VAR"` (or `"${VAR}"` but that's totally unnecessary)

**Comment by gaborh-da** [comment]
> and please attend the warnings about Bash, e.g. replace all `${VAR}` with `"$VAR"` (or `"${VAR}"` but that's totally unnecessary)

what else?

**Review by richardkapolnai-da — APPROVED** [review]

**Review by richardkapolnai-da — DISMISSED** [review]

File: scripts/dabl_deploy.sh (hunk 3 above)
Comment (gaborh-da) [reply scripts/dabl_deploy.sh]: removed the comment

**Comment by gaborh-da** [comment]
I tried:
- on DABL with the uploader
- locally, with Canton
- locally, with Sandbox
I followed the README.

**Review by richardkapolnai-da — APPROVED** [review]

**Pull request merged**